supabase/migrations/20240611113313_rpc_add_update_onboarding_status.sql

```diff
@@ -0,0 +1,22 @@
+/*
+  This migration defines a function to update the onboarding status fields on both the stripe_users and service_providers tables.
+  Updates to both rows are implicitly wrapped in a transaction, so if one update fails, the other will be rolled back.
+
+  The main onboarding source of truth is on the stripe_users table. This is used to flag a user's onboarding status with Stripe.
+
+  On the service_providers table, we use an indexed onboarding_status field to hide non-onboarded providers from our search results.
+*/
+
+create or replace function update_service_provider_onboarding(p_user_id uuid, p_onboarding_status boolean)
+returns void as $$
+begin
+  update stripe_users
+  set onboarded = p_onboarding_status
+  where user_id = p_user_id and type = 'SERVICE_PROVIDER';
+
+  update service_providers
+  set onboarding_status = p_onboarding_status
+  where user_id = p_user_id;
+end;
+$$ language plpgsql;
+
```

Comment:
Here we are using RPCs as implicit SQL transactions to write to two tables. This allows us to "all or nothing" update so we can expect both updates to succeed, or both to fail. There is a performance cost though, so weigh out your pros and cons.

Flagging this so I can reference this in the future.